### PR TITLE
feat(1495): As UX lead, I want the user to have a organized first experience for the Profile page

### LINKED ISSUE #1495: As UX lead, I want the user to have an organized first experience for the Profile page
**Description:**
Per #653 , we are updating our UI for the Profile page for users requesting access.

**Acceptance Criteria:** 
- [ ] Any page that's visible in the nav bar should be within the users' _current_ permissions (e.g. If you don’t have a role you don’t get the ‘Data Files’ or ‘Admin’ nav items—If you haven’t requested access yet then you shouldn’t see ‘Profile’)
- [ ] Access request form should exist on “Home”, not “Profile”.
- [ ] Users should only submit request access form ONCE. Once submitted, their homepage content should be the status message confirming that they requested access. Once granted access, their homepage content should reflect their role/permissions in TDP.
- [ ] Once a request for access has been submitted, users should have access to their profile page
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
- [x] User is shown "request for access pending message until approved per Figma link below
- [x] User is shown their intake infomation as well
- [x] A11y review must be completed
- [x] Implement backend request access logic
- [ ] Will not show "STT analyst" role back to approved user, will only show OFA roles for approved users
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**

**Supporting Documentation:**
- [Figma Mock Ups](https://www.figma.com/file/irgQPLTrajxCXNiYBTEnMV/TDP-Mockups-For-Feedback?node-id=5892%3A48657)
- [Video Demo](https://user-images.githubusercontent.com/28515874/146063268-290ae9d2-9fd9-4db8-96cd-a7dae9169972.mp4 "Video Demo")

**Open Questions:**

### PR DESCRIPTION
**Pull request by jorgegonzalez**
## Summary of Changes

This PR firstly adds a new `access_request` concept on the user model with which we use to manage the pre and post-approval user states on the frontend. This includes major refactors to the Home and Profile pages to improve the access request flow for new users. Additionally the `/welcome` route is now `/home`, `/edit-profile` is now `/profile`, and `/request` has been deprecated.

Pull request closes #1495

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Go through the list of ACs and ensure everything works as expected.
> *Demo GIF(s) and screenshots for testing procedure*

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] Any page that's visible in the nav bar should be within the users' _current_ permissions (e.g. If you haven’t requested access yet then you shouldn’t see ‘Profile’.  If you don’t have a role you don’t get the ‘Data Files’ or ‘Admin’ nav items)
- [x] Access request form should exist on “Home”, not “Profile”.
- [x] Users should only submit request access form ONCE. Once submitted, their homepage content should be the status message confirming that they requested access. Once granted access, their homepage content should reflect their role/permissions in TDP.
- [x] Once a request for access has been submitted, users should have access to their profile page
- [x] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [x] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [x] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/tdpservice/users/admin.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ class UserAdmin(admin.ModelAdmin):
     exclude = ['password', 'user_permissions', 'is_active']
     readonly_fields = ['last_login', 'date_joined', 'login_gov_uuid', 'hhs_id']
     form = UserForm
+    list_filter = ('access_request',)
```
Comment (jorgegonzalez) [inline tdrs-backend/tdpservice/users/admin.py:26]: @raftmsohani added this list filter so admins can easily see a list of user's pending access approval

File: tdrs-backend/tdpservice/users/migrations/0028_user_access_request.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('users', '0027_user_hhs_id'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='user',
+            name='access_request',
```
Comment (jorgegonzalez) [inline tdrs-backend/tdpservice/users/migrations/0028_user_access_request.py:15]: Related migration for the `access_request` field.

File: tdrs-backend/tdpservice/users/views.py [hunk 3]
```diff
@@ -51,6 +51,18 @@ def set_profile(self, request, pk=None):
         )
         return Response(serializer.data)
 
+    @action(methods=["PATCH"], detail=False)
+    def request_access(self, request, pk=None):
```
Comment (jorgegonzalez) [inline tdrs-backend/tdpservice/users/views.py:55]: @raftmsohani added this new request_access endpoint for the `access_request` field to be toggled from the frontend.

File: tdrs-frontend/src/components/Header/Header.jsx [hunk 4]
```diff
@@ -108,22 +109,22 @@ function HeaderComp() {
...
-                    tabTitle="Data Files"
-                    href="/data-files"
-                  />
-                  <NavItem
-                    pathname={pathname}
-                    tabTitle="Profile"
-                    href="/edit-profile"
-                  />
-                  {isOFASystemAdmin() && (
+                  <NavItem pathname={pathname} tabTitle="Home" href="/home" />
```
Comment (jorgegonzalez) [inline tdrs-frontend/src/components/Header/Header.jsx:112]: Hiding additional navitems based on the access request status and user roles

File: tdrs-frontend/src/components/Home/Home.jsx [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+  }
+  return null
+}
+
+/**
+ * Home renders the Request Access form for creating a profile, and displays
+ * a pending-approval state, before showing the user their active roles and
+ * permissions once they are approved by an Admin in the backend.
+ */
+function Home() {
```
Comment (jorgegonzalez) [inline tdrs-frontend/src/components/Home/Home.jsx:45]: This file is mostly the same as the old Profile page, with additional states handling the transitions based on the access request status.

File: tdrs-frontend/src/components/Routes/Routes.js [hunk 6]
```diff
@@ -23,17 +22,14 @@ const Routes = () => {
...
+        <Home />
       </PrivateRoute>
       <PrivateRoute exact title="TANF Data Files" path="/data-files">
         <Reports />
       </PrivateRoute>
-      <PrivateRoute exact title="Request Access" path="/edit-profile">
-        <EditProfile />
-      </PrivateRoute>
-      <PrivateRoute exact title="Request Submitted" path="/request">
-        <Request />
```
Comment (jorgegonzalez) [inline tdrs-frontend/src/components/Routes/Routes.js:36]: As this is captured in the new Home page, this route and component are deprecated.

**Comment by abottoms-coder** [comment]
@jorgegonzalez Just some house-keeping questions:

- Can you please update the description with ACs and testing steps when you're ready?
- A comparison before/after GIF would be awesome!
- Is the column status "in progress" accurate or would you like this moved into the raft review column? If so, happy to start reviewing.

File: tdrs-backend/tdpservice/users/serializers.py [hunk 7]
```diff
@@ -70,7 +71,7 @@ class Meta:
         """Metadata."""
 
         model = User
-        fields = ['id', 'first_name', 'last_name', 'email', 'stt', 'roles']
+        fields = ['id', 'first_name', 'last_name', 'email', 'stt', 'roles', 'access_request']
```
Comment (raftmsohani) [inline tdrs-backend/tdpservice/users/serializers.py:74]: to ensure access_request is listed in user API endpoint

**Review by raftmsohani — APPROVED** [review]

**Comment by jorgegonzalez** [comment]
- [ ] Add access request field description

File: tdrs-backend/tdpservice/users/admin.py (hunk 1 above)
Comment (jorgegonzalez) [reply tdrs-backend/tdpservice/users/admin.py:26]: The flag will never be removed, so it will have to look at the users with no roles.

**Comment by jorgegonzalez** [comment]
This PR is awaiting a final a11y review before moving to QASP.

**Comment by reitermb** [comment]
A11y is looking good!
Video showing new flow behavior is attached. Info banner is successfully announced to screenreaders when present on the profile page.

https://user-images.githubusercontent.com/28515874/153267385-e69cf382-e5e4-46cb-86d3-e923ad9f8aaf.mp4

**Review by abottoms-coder — APPROVED** [review]
LGTM locally and in raft-space. Was a bit confused that this didn't seem to deploy to the QASP space but I think this was just some label confusion.

File: tdrs-backend/tdpservice/users/admin.py (hunk 1 above)
Comment (ADPennington) [reply tdrs-backend/tdpservice/users/admin.py:26]: @jorgegonzalez what does it mean for [Miles' account](https://tdp-backend-raft.app.cloud.gov/admin/users/user/d44233b4-82f4-4e7d-b5bd-f28e3f65732e/change/?_changelist_filters=access_request__exact%3D1) to have a green check while also having roles already assigned? (note: i didn't modify this account as part of this review).

**Comment by ttran-hub** [comment]
> @ttran-hub this is ready for `gov a11y review`. the link to use for testing is [here](https://tdp-frontend-raft.app.cloud.gov/) and raft's a11y review is [here](https://github.com/raft-tech/TANF-app/pull/1571#issuecomment-1034075090)
> 
> cc: @iamjolly

LGMT.  there's an instance (href="/" title="Home" aria-label="Home">TANF Data Portal</a>) that trigger the title and aria-label mismatch but not in scope of this pull request.   Flag for future discussion

**Comment by ADPennington** [comment]
> > @ttran-hub this is ready for `gov a11y review`. the link to use for testing is [here](https://tdp-frontend-raft.app.cloud.gov/) and raft's a11y review is [here](https://github.com/raft-tech/TANF-app/pull/1571#issuecomment-1034075090)
> > cc: @iamjolly
> 
> LGMT. there's an instance (href="/" title="Home" aria-label="Home">TANF Data Portal) that trigger the title and aria-label mismatch but not in scope of this pull request. Flag for future discussion

thank you @ttran-hub 🚀 

@reitermb can you add this to your list of items to follow-up on? i'd like to get this on the record prior to approval.

**Comment by reitermb** [comment]
> > > @ttran-hub this is ready for `gov a11y review`. the link to use for testing is [here](https://tdp-frontend-raft.app.cloud.gov/) and raft's a11y review is [here](https://github.com/raft-tech/TANF-app/pull/1571#issuecomment-1034075090)
> > > cc: @iamjolly
> > 
> > 
> > LGMT. there's an instance (href="/" title="Home" aria-label="Home">TANF Data Portal) that trigger the title and aria-label mismatch but not in scope of this pull request. Flag for future discussion
> 
> thank you @ttran-hub 🚀
> 
> @reitermb can you add this to your list of items to follow-up on? i'd like to get this on the record prior to approval.

I've spun up a kind of a11y spike for Thomas and I to sync on and make into an actionable fix! #1646

**Comment by lfrohlich** [comment]
Thanks @reitermb . With that, good to approve

**Pull request merged**